client.go

```diff
@@ -24,23 +23,41 @@ func NewClient(config *rest.Config) (mf.Client, error) {
 	if err != nil {
 		return nil, err
 	}
+
 	mapper, err := apiutil.NewDynamicRESTMapper(config)
+
 	if err != nil {
 		return nil, err
 	}
-	return &clientGoClient{client: client, mapper: mapper}, nil
+	resourceGetter := &clientRestMapper{
+		client: client,
+		mapper: mapper,
+	}
+
+	return &clientGoClient{client: client, resourceGetter: resourceGetter}, nil
+}
+
+func NewDynamicClient(client dynamic.Interface) (mf.Client, error) {
+	resourceGetter := &unsafeResourceInterface{
+		client: client,
+	}
+	return &clientGoClient{client: client, resourceGetter: resourceGetter}, nil
+}
+
+type resourceGetter interface {
+	resourceInterface(obj *unstructured.Unstructured) (dynamic.ResourceInterface, error)
 }
 
 type clientGoClient struct {
-	client dynamic.Interface
-	mapper meta.RESTMapper
+	client         dynamic.Interface
```

Comment:
This doesn't seem to be used anymore. Because the `resourceGetter` impl's get their own `dynamic.Interface`, the `clientGoClient` no longer needs it.

Reply:
Nice. Done.